tests: `-Copt-level=3` instead of `-O` in codegen tests

workingjubilee opened the pull request:
An effective blocker for redefining the meaning of `-O` is to stop reusing this somewhat ambiguous alias in our own codegen test suite. The choice between `-Copt-level=2` and `-Copt-level=3` is arbitrary for most of our tests. In most cases it makes no difference, so I set most of them to `-Copt-level=3`, as it will lead to slightly more "normalized" codegen.

try-job: test-various
try-job: arm-android
try-job: armhf-gnu
try-job: i686-gnu-1
try-job: i686-gnu-2
try-job: i686-mingw
try-job: i686-msvc-1
try-job: i686-msvc-2
try-job: aarch64-apple
try-job: aarch64-gnu

workingjubilee committed:
tests/codegen: use -Copt-level=3 instead of -O

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit ba88cad33de26fa5578eb49e580fa0679502d79e with merge 88aac115faa8c49b17153e74966b1b163f14ad12...
<!-- homu: {"type":"TryBuildStarted","head_sha":"ba88cad33de26fa5578eb49e580fa0679502d79e","merge_sha":"88aac115faa8c49b17153e74966b1b163f14ad12"} -->

rust-log-analyzer commented:
The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/13222411236/job/36909132785) [(plain)](https://github.com/rust-lang-ci/rust/commit/88aac115faa8c49b17153e74966b1b163f14ad12/checks/36909132785/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
test [codegen] tests/codegen/zst-offset.rs ... ok

failures:

---- [codegen] tests/codegen/dont-shufflevector-bswaps.rs#OPT3 stdout ----

error in revision `OPT3`: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll" "/checkout/tests/codegen/dont-shufflevector-bswaps.rs" "--check-prefix=CHECK" "--check-prefix" "OPT3" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/dont-shufflevector-bswaps.rs:15:10: error: OPT3: expected string not found in input
// OPT3: load <8 x i16>
         ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:7:32: note: scanning from here
define dso_local void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 {
                               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:9:8: note: possible intended match here
 %_4 = load i16, ptr %value, align 2, !noundef !1


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll
Check file: /checkout/tests/codegen/dont-shufflevector-bswaps.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'dont_shufflevector_bswaps.1504e1bca058720f-cgu.0' 
            2: source_filename = "dont_shufflevector_bswaps.1504e1bca058720f-cgu.0" 
            3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-n32:64-S128-ni:1:10:20" 
            5:  
            5:  
            6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) 
            7: define dso_local void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 { 
check:15'0                                    X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
check:15'0     ~~~~~~~
check:15'0     ~~~~~~~
            9:  %_4 = load i16, ptr %value, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:15'1            ?                                           possible intended match
           10:  %0 = tail call i16 @llvm.bswap.i16(i16 %_4) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           11:  %1 = getelementptr inbounds i8, ptr %value, i32 2 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           12:  %_6 = load i16, ptr %1, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           13:  %2 = tail call i16 @llvm.bswap.i16(i16 %_6) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           14:  %3 = getelementptr inbounds i8, ptr %value, i32 4 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           15:  %_8 = load i16, ptr %3, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           16:  %4 = tail call i16 @llvm.bswap.i16(i16 %_8) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           17:  %5 = getelementptr inbounds i8, ptr %value, i32 6 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           18:  %_10 = load i16, ptr %5, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           19:  %6 = tail call i16 @llvm.bswap.i16(i16 %_10) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           20:  %7 = getelementptr inbounds i8, ptr %value, i32 8 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           21:  %_12 = load i16, ptr %7, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           22:  %8 = tail call i16 @llvm.bswap.i16(i16 %_12) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           23:  %9 = getelementptr inbounds i8, ptr %value, i32 10 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           24:  %_14 = load i16, ptr %9, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           25:  %10 = tail call i16 @llvm.bswap.i16(i16 %_14) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           26:  %11 = getelementptr inbounds i8, ptr %value, i32 12 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           27:  %_16 = load i16, ptr %11, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           28:  %12 = tail call i16 @llvm.bswap.i16(i16 %_16) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           29:  %13 = getelementptr inbounds i8, ptr %value, i32 14 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           30:  %_18 = load i16, ptr %13, align 2, !noundef !1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           31:  %14 = tail call i16 @llvm.bswap.i16(i16 %_18) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           32:  store i16 %0, ptr %_0, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           33:  %addr16.sroa.2.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 2 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34:  store i16 %2, ptr %addr16.sroa.2.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35:  %addr16.sroa.3.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 4 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           36:  store i16 %4, ptr %addr16.sroa.3.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           37:  %addr16.sroa.4.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 6 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           38:  store i16 %6, ptr %addr16.sroa.4.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           39:  %addr16.sroa.5.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 8 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           40:  store i16 %8, ptr %addr16.sroa.5.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           41:  %addr16.sroa.6.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 10 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           42:  store i16 %10, ptr %addr16.sroa.6.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43:  %addr16.sroa.7.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 12 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           44:  store i16 %12, ptr %addr16.sroa.7.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           45:  %addr16.sroa.8.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 14 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46:  store i16 %14, ptr %addr16.sroa.8.0._0.sroa_idx, align 1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           47:  ret void 
check:15'0     ~~~~~~~~~~
           48: } 
check:15'0     ~~
check:15'0     ~
check:15'0     ~
           50: ; Function Attrs: mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           51: declare i16 @llvm.bswap.i16(i16) #1 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:15'0     ~
check:15'0     ~
           53: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) "target-cpu"="generic" } 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           54: attributes #1 = { mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:15'0     ~
check:15'0     ~
           56: !llvm.ident = !{!0} 
check:15'0     ~~~~~~~~~~~~~~~~~~~~
check:15'0     ~
check:15'0     ~
           58: !0 = !{!"rustc version 1.86.0-nightly (88aac115f 2025-02-09)"} 
check:15'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           59: !1 = !{} 
check:15'0     ~~~~~~~~~
------------------------------------------



```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/13222411236/job/36909812274)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/13222411236/job/36909812274","builder_name":"checks-actions"} -->

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit bba98a6ac0f53328aca4408171cd1be67f816b48 with merge 19ec3e24172e1433463f9a8079805a5d43daa8f8...
<!-- homu: {"type":"TryBuildStarted","head_sha":"bba98a6ac0f53328aca4408171cd1be67f816b48","merge_sha":"19ec3e24172e1433463f9a8079805a5d43daa8f8"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13222757508/job/36909848953) [(plain)](https://github.com/rust-lang/rust/commit/bba98a6ac0f53328aca4408171cd1be67f816b48/checks/36909848953/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#21 exporting to docker image format
#21 sending tarball 29.3s done
#21 DONE 31.8s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---


failures:

---- [codegen] tests/codegen/dont-shufflevector-bswaps.rs#OPT3 stdout ----

error in revision `OPT3`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-18/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll" "/checkout/tests/codegen/dont-shufflevector-bswaps.rs" "--check-prefix=CHECK" "--check-prefix" "OPT3" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/dont-shufflevector-bswaps.rs:16:15: error: OPT3-NEXT: expected string not found in input
// OPT3-NEXT: call <8 x i16> @llvm.bswap
              ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:11:17: note: scanning from here
 store <8 x i16> %1, ptr %_0, align 1
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:16:4: note: possible intended match here
declare <8 x i16> @llvm.bswap.v8i16(<8 x i16>) #1


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll
Check file: /checkout/tests/codegen/dont-shufflevector-bswaps.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
           1: ; ModuleID = 'dont_shufflevector_bswaps.1504e1bca058720f-cgu.0' 
           2: source_filename = "dont_shufflevector_bswaps.1504e1bca058720f-cgu.0" 
           3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
           5:  
           5:  
           6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: readwrite) uwtable 
           7: define void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 { 
           8: start: 
           9:  %0 = load <8 x i16>, ptr %value, align 2 
          10:  %1 = tail call <8 x i16> @llvm.bswap.v8i16(<8 x i16> %0) 
          11:  store <8 x i16> %1, ptr %_0, align 1 
next:16'0                     X~~~~~~~~~~~~~~~~~~~~~ error: no match found
          12:  ret void 
next:16'0     ~~~~~~~~~~
          13: } 
next:16'0     ~~
next:16'0     ~
next:16'0     ~
          15: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          16: declare <8 x i16> @llvm.bswap.v8i16(<8 x i16>) #1 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:16'1        ?                                               possible intended match
next:16'0     ~
next:16'0     ~
          18: attributes #0 = { mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          19: attributes #1 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:16'0     ~
next:16'0     ~
          21: !llvm.module.flags = !{!0, !1} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          22: !llvm.ident = !{!2} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~
next:16'0     ~
next:16'0     ~
          24: !0 = !{i32 8, !"PIC Level", i32 2} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          25: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          26: !2 = !{!"rustc version 1.86.0-nightly (72e0fb7c8 2025-02-09)"} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit daf1ffc3a2e74ea60ad9a4a415ed040f46e7d8ed with merge 8c43bfa711d64dd532f82e0cdadcaa6c0d50d8a8...
<!-- homu: {"type":"TryBuildStarted","head_sha":"daf1ffc3a2e74ea60ad9a4a415ed040f46e7d8ed","merge_sha":"8c43bfa711d64dd532f82e0cdadcaa6c0d50d8a8"} -->

rust-log-analyzer commented:
The job **`aarch64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/13222767306/job/36909865945) [(plain)](https://github.com/rust-lang-ci/rust/commit/19ec3e24172e1433463f9a8079805a5d43daa8f8/checks/36909865945/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
test [codegen] tests/codegen/zip.rs ... ok

failures:

---- [codegen] tests/codegen/dont-shufflevector-bswaps.rs#OPT3 stdout ----

error in revision `OPT3`: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/aarch64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll" "/checkout/tests/codegen/dont-shufflevector-bswaps.rs" "--check-prefix=CHECK" "--check-prefix" "OPT3" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/dont-shufflevector-bswaps.rs:16:15: error: OPT3-NEXT: expected string not found in input
// OPT3-NEXT: call <8 x i16> @llvm.bswap
              ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:11:17: note: scanning from here
 store <8 x i16> %1, ptr %_0, align 1
                ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll:16:4: note: possible intended match here
declare <8 x i16> @llvm.bswap.v8i16(<8 x i16>) #1


Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen/dont-shufflevector-bswaps.OPT3/dont-shufflevector-bswaps.ll
Check file: /checkout/tests/codegen/dont-shufflevector-bswaps.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
           1: ; ModuleID = 'dont_shufflevector_bswaps.1504e1bca058720f-cgu.0' 
           2: source_filename = "dont_shufflevector_bswaps.1504e1bca058720f-cgu.0" 
           3: target datalayout = "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
           5:  
           5:  
           6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
           7: define void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 { 
           8: start: 
           9:  %0 = load <8 x i16>, ptr %value, align 2 
          10:  %1 = tail call <8 x i16> @llvm.bswap.v8i16(<8 x i16> %0) 
          11:  store <8 x i16> %1, ptr %_0, align 1 
next:16'0                     X~~~~~~~~~~~~~~~~~~~~~ error: no match found
          12:  ret void 
next:16'0     ~~~~~~~~~~
          13: } 
next:16'0     ~~
next:16'0     ~
next:16'0     ~
          15: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          16: declare <8 x i16> @llvm.bswap.v8i16(<8 x i16>) #1 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:16'1        ?                                               possible intended match
next:16'0     ~
next:16'0     ~
          18: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          19: attributes #1 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:16'0     ~
next:16'0     ~
          21: !llvm.module.flags = !{!0} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          22: !llvm.ident = !{!1} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~
next:16'0     ~
next:16'0     ~
          24: !0 = !{i32 8, !"PIC Level", i32 2} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          25: !1 = !{!"rustc version 1.86.0-nightly (19ec3e241 2025-02-09)"} 
next:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit dd1ee8c917a1c44b1892bb514d53c456ae82e7e0 with merge f1e2ad852d6aa6412ad042f508d0ecf19500bd26...
<!-- homu: {"type":"TryBuildStarted","head_sha":"dd1ee8c917a1c44b1892bb514d53c456ae82e7e0","merge_sha":"f1e2ad852d6aa6412ad042f508d0ecf19500bd26"} -->

workingjubilee committed:
tests: issue-122805 -> dont-shuffle-bswaps

workingjubilee committed:
tests: simplify dont-shuffle-bswaps test

This should guarantee it tests what we want it to test and no more.
It should probably also run on 64-bit platforms that are not x86-64,
which will often have the vector registers the opt implies.

saethlin commented on tests/codegen/dont-shuffle-bswaps.rs:
```diff
@@ -0,0 +1,36 @@
+//@ revisions: OPT2 OPT3
+//@[OPT2] compile-flags: -Copt-level=2
```
It's a bit odd to not have OPT2 filecheck comments, but I guess the idea here is to check for the lack of a shufflevector at level 2, and check that we get the slick bswap codegen at level 3?

workingjubilee replied:
yes, we're checking for "no shufflevector plz" at both levels and then for the bswap at OPT3

workingjubilee replied:
for OPT3 the only way it could have a shufflevector instruction is if it smuggled itself in right between `start:` and the actual checked lines, but, idk, might as well?

saethlin commented:
@bors r+

In spite of this touching every codegen test I bet it's actually not going to conflict. Except for one test, this is just changing the flags line.

bors commented:
:pushpin: Commit 2ea795337c956b6841ef4b91cc09c1551ecc7c13 has been approved by `saethlin`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=saethlin 2ea795337c956b6841ef4b91cc09c1551ecc7c13 -->
<!-- homu: {"type":"Approved","sha":"2ea795337c956b6841ef4b91cc09c1551ecc7c13","approver":"saethlin","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

workingjubilee commented:
yeah the only reason I put the assembly and codegen tests in different PRs was to make it faster to iterate on the try-jobs for both sets at the same time.

jieyouxu commented:
Failed in rollup: https://github.com/rust-lang/rust/pull/136850#issuecomment-2650356818

```
---- [codegen] tests/codegen/dont-shuffle-bswaps.rs#OPT3 stdout ----

error in revision `OPT3`: verification with 'FileCheck' failed
status: exit status: 1
[...]
/checkout/tests/codegen/dont-shuffle-bswaps.rs:15:10: error: OPT3: expected string not found in input
// OPT3: load <8 x i16>
         ^
```

@bors r-

jieyouxu commented on tests/codegen/dont-shuffle-bswaps.rs:
```diff
@@ -0,0 +1,36 @@
+// to avoid complicating the code.
+// CHECK-LABEL: define{{.*}}void @convert(
+// CHECK-NOT: shufflevector
+// On higher opt levels, this should just be a bswap:
+// OPT3: load <8 x i16>
+// OPT3-NEXT: call <8 x i16> @llvm.bswap
+// OPT3-NEXT: store <8 x i16>
+// OPT3-NEXT: ret void
```
This `load` is missing (or different) on `test-various` at `-O3`

workingjubilee commented:
yeah some targets don't do the opt, lol

workingjubilee commented:
@bors try

bors commented:
:hourglass: Trying commit d76bcf2d005da18b8418c85629b5f0f72076cada with merge e1f698a54b8f1325fef15f91f118a6c61a253e5a...
<!-- homu: {"type":"TryBuildStarted","head_sha":"d76bcf2d005da18b8418c85629b5f0f72076cada","merge_sha":"e1f698a54b8f1325fef15f91f118a6c61a253e5a"} -->

rust-log-analyzer commented:
The job **`armhf-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/13261175656/job/37017928646) [(plain)](https://github.com/rust-lang-ci/rust/commit/e1f698a54b8f1325fef15f91f118a6c61a253e5a/checks/37017928646/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
test [codegen] tests/codegen/zip.rs ... ok

failures:

---- [codegen] tests/codegen/dont-shuffle-bswaps.rs#OPT3 stdout ----

error in revision `OPT3`: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shuffle-bswaps.OPT3/dont-shuffle-bswaps.ll" "/checkout/tests/codegen/dont-shuffle-bswaps.rs" "--check-prefix=CHECK" "--check-prefix" "OPT3" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/dont-shuffle-bswaps.rs:17:10: error: OPT3: expected string not found in input
// OPT3: load <8 x i16>
         ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shuffle-bswaps.OPT3/dont-shuffle-bswaps.ll:7:22: note: scanning from here
define void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 {
                     ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shuffle-bswaps.OPT3/dont-shuffle-bswaps.ll:9:8: note: possible intended match here
 %_4 = load i16, ptr %value, align 2, !noundef !2


Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/dont-shuffle-bswaps.OPT3/dont-shuffle-bswaps.ll
Check file: /checkout/tests/codegen/dont-shuffle-bswaps.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'dont_shuffle_bswaps.5c93dbc6d133cf57-cgu.0' 
            2: source_filename = "dont_shuffle_bswaps.5c93dbc6d133cf57-cgu.0" 
            3: target datalayout = "e-m:e-p:32:32-Fi8-i64:64-v128:64:128-a:0:32-n32-S64" 
            4: target triple = "arm-unknown-linux-gnueabihf" 
            5:  
            6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable 
            7: define void @convert(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([16 x i8]) align 1 dereferenceable(16) %_0, ptr noalias nocapture noundef readonly align 2 dereferenceable(16) %value) unnamed_addr #0 { 
check:17'0                          X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
check:17'0     ~~~~~~~
check:17'0     ~~~~~~~
            9:  %_4 = load i16, ptr %value, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:17'1            ?                                           possible intended match
           10:  %0 = tail call i16 @llvm.bswap.i16(i16 %_4) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           11:  %1 = getelementptr inbounds i8, ptr %value, i32 2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           12:  %_6 = load i16, ptr %1, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           13:  %2 = tail call i16 @llvm.bswap.i16(i16 %_6) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           14:  %3 = getelementptr inbounds i8, ptr %value, i32 4 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           15:  %_8 = load i16, ptr %3, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           16:  %4 = tail call i16 @llvm.bswap.i16(i16 %_8) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           17:  %5 = getelementptr inbounds i8, ptr %value, i32 6 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           18:  %_10 = load i16, ptr %5, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           19:  %6 = tail call i16 @llvm.bswap.i16(i16 %_10) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           20:  %7 = getelementptr inbounds i8, ptr %value, i32 8 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           21:  %_12 = load i16, ptr %7, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           22:  %8 = tail call i16 @llvm.bswap.i16(i16 %_12) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           23:  %9 = getelementptr inbounds i8, ptr %value, i32 10 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           24:  %_14 = load i16, ptr %9, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           25:  %10 = tail call i16 @llvm.bswap.i16(i16 %_14) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           26:  %11 = getelementptr inbounds i8, ptr %value, i32 12 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           27:  %_16 = load i16, ptr %11, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           28:  %12 = tail call i16 @llvm.bswap.i16(i16 %_16) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           29:  %13 = getelementptr inbounds i8, ptr %value, i32 14 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           30:  %_18 = load i16, ptr %13, align 2, !noundef !2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           31:  %14 = tail call i16 @llvm.bswap.i16(i16 %_18) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           32:  store i16 %0, ptr %_0, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           33:  %addr16.sroa.2.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 2 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           34:  store i16 %2, ptr %addr16.sroa.2.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           35:  %addr16.sroa.3.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 4 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           36:  store i16 %4, ptr %addr16.sroa.3.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           37:  %addr16.sroa.4.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 6 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           38:  store i16 %6, ptr %addr16.sroa.4.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           39:  %addr16.sroa.5.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 8 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           40:  store i16 %8, ptr %addr16.sroa.5.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           41:  %addr16.sroa.6.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 10 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           42:  store i16 %10, ptr %addr16.sroa.6.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43:  %addr16.sroa.7.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 12 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           44:  store i16 %12, ptr %addr16.sroa.7.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           45:  %addr16.sroa.8.0._0.sroa_idx = getelementptr inbounds i8, ptr %_0, i32 14 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46:  store i16 %14, ptr %addr16.sroa.8.0._0.sroa_idx, align 1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           47:  ret void 
check:17'0     ~~~~~~~~~~
           48: } 
check:17'0     ~~
check:17'0     ~
check:17'0     ~
           50: ; Function Attrs: mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           51: declare i16 @llvm.bswap.i16(i16) #1 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:17'0     ~
check:17'0     ~
           53: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: readwrite) uwtable "target-cpu"="generic" "target-features"="+strict-align,+v6,+vfp2,-d32" } 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           54: attributes #1 = { mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:17'0     ~
check:17'0     ~
           56: !llvm.module.flags = !{!0} 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
           57: !llvm.ident = !{!1} 
check:17'0     ~~~~~~~~~~~~~~~~~~~~
check:17'0     ~
check:17'0     ~
           59: !0 = !{i32 8, !"PIC Level", i32 2} 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           60: !1 = !{!"rustc version 1.86.0-nightly (e1f698a54 2025-02-11)"} 
check:17'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           61: !2 = !{} 
check:17'0     ~~~~~~~~~
------------------------------------------



```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/13261175656/job/37021142342)
<!-- homu: {"type":"TryBuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/13261175656/job/37021142342","builder_name":"checks-actions"} -->

workingjubilee commented:
think I have a better idea...

@bors try

bors commented:
:hourglass: Trying commit 16fb1c452090c471fd2c45936b7402225d511a93 with merge a28e7fa9603d6c4f39773a02f028cc4c0dc65108...
<!-- homu: {"type":"TryBuildStarted","head_sha":"16fb1c452090c471fd2c45936b7402225d511a93","merge_sha":"a28e7fa9603d6c4f39773a02f028cc4c0dc65108"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/13270256589/job/37056592453)
Build commit: a28e7fa9603d6c4f39773a02f028cc4c0dc65108 (`a28e7fa9603d6c4f39773a02f028cc4c0dc65108`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/13270256589/job/37056592453"},"merge_sha":"a28e7fa9603d6c4f39773a02f028cc4c0dc65108"} -->

workingjubilee commented:
@bors r=saethlin

bors commented:
:pushpin: Commit 3c0c9b6770f1ac0324be91580f52b6f45346e47b has been approved by `saethlin`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=saethlin 3c0c9b6770f1ac0324be91580f52b6f45346e47b -->
<!-- homu: {"type":"Approved","sha":"3c0c9b6770f1ac0324be91580f52b6f45346e47b","approver":"saethlin","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

workingjubilee commented:
Nah, that error looks like the stdarch update. Looks like `cg_clif` stopped compiling some code in stdarch due to updates to the Arm intrinsic generator?

GuillaumeGomez commented:
Noted, thanks! Gonna comment there then!
